### DISCUSSION TITLE
How to register listener using WorkflowStatusListener interface of OSS/Netflix Conductor

### DISCUSSION THREAD

**Discussion by PaulPheonix06** [conductor-oss/conductor-community#8] [Q&A]
Custom listener registration for Netflix Conductor to trigger upon workflow completion but not functioning as expected.

How can I register a custom listener in my code that should trigger when Netflix Conductor finishes executing a workflow? It seems that the listener is not being invoked upon workflow completion, even though I have set workflowStatusListenerEnabled to true. Am i missing anything?

Below is the code for my Custom Listener :

`import com.netflix.conductor.common.run.Workflow;
import com.netflix.conductor.common.run.Workflow.WorkflowStatus;
import com.netflix.conductor.core.listener.WorkflowStatusListener;
import com.netflix.conductor.model.WorkflowModel;

import org.springframework.stereotype.Component;

@Component
public class MyWorkflowStatusListener implements WorkflowStatusListener {

    @Override
    public void onWorkflowCompleted(WorkflowModel workflow) {
        System.out.println("Completed" + workflow);
    }

    @Override
    public void onWorkflowTerminated(WorkflowModel workflow) {
        System.out.println("Terminated" + workflow);
    }

    @Override
    public void onWorkflowFinalized(WorkflowModel workflow) {
        System.out.println("finalized" + workflow);
    }
}`

**Comment by fcl-9** [comment]
Hey @PaulPheonix06 did you come to any conclusion on this? I have found this PR referenced from the official docs https://github.com/Netflix/conductor/issues/1017#issuecomment-468869173 but it is not clear what needs to be done to enable it?

Update:
In order to enable a Workflow listener we need to set the workflowStatusListenerEnabled field in your workflow definition to true which enables notifications. Then we either use an existing WorkflowStatusListener (i.e. `StatusChangePublisher.java`, `ConductorQueueStatusPublisher.java`, `ArchivingWorkflowStatusListener.java` classes). Also, it is possible to create our own Workflow Status Listener, in order to do it we need to extend the following interface WorkflowStatusListener. This WorkflowStatusListener can be implemented in such a way that it can send an event to conductor queue or communicate with external systems.

In order to enable of the existing WorkflowListener we need to set the following configuration `conductor.workflow-status-listener.type` to archive, queue_publisher or workflow_publisher. Different WorkflowListeners may need extra configurations as the ConductorQueueStatusPublisher.

Hope this helps someone else :)
